src/filesystem_helper.cpp

```diff
@@ -306,6 +306,44 @@ path temp_directory_path()
   return path(temp_path);
 }
 
+path create_temp_directory(const std::string & base_name, const path & parent_path)
+{
+  const auto template_path = base_name + "XXXXXX";
+  std::string full_template_str = (parent_path / template_path).string();
+  std::vector<char> mutable_full_template_data(full_template_str.begin(), full_template_str.end());
```

Comment:
BTW your original code was OK too, as @clalancette pointed out in https://github.com/ros2/rcpputils/pull/126#discussion_r577930719. 

My apologies for the unnecessary noise :facepalm: